javascript/chronometer.js

```diff
@@ -1,36 +1,120 @@
-class Chronometer {
+/*class Chronometer {
   constructor() {
-    // ... your code goes here
+    this.currentTime = 0;
+    this.intervalId = null;
   }
 
   start(printTimeCallback) {
-    // ... your code goes here
+    this.intervalId = setInterval(() => {
+      this.currentTime++;
+      if (typeof printTimeCallback === "function") {
+        printTimeCallback();
+      };
+    },1000)
   }
 
   getMinutes() {
-    // ... your code goes here
+    return Math.floor(this.currentTime/60);
   }
 
   getSeconds() {
-    // ... your code goes here
+    return this.currentTime%60;
   }
 
   computeTwoDigitNumber(value) {
-    // ... your code goes here
+    if (value === 0) {
+      return "00"
+    }
+    else if (value.toString().length === 1) {
+      return `0${value}`
+    }  
+    else {
+      return value.toString()
+    };
   }
 
   stop() {
-    // ... your code goes here
+    clearInterval(this.intervalId);
   }
 
   reset() {
-    // ... your code goes here
+    this.currentTime = 0;
   }
 
   split() {
-    // ... your code goes here
+    return `${this.computeTwoDigitNumber(this.getMinutes())}:${this.computeTwoDigitNumber(this.getSeconds())}`
   }
 }
+*/
+
+// ******* Iteration 9 *********
+
+ class Chronometer {
+  constructor() {
+    this.currentTime = 0;
+    this.intervalId = null;
+    this.milliseconds = 0;
+    this.millisecondsId = null;
+  }
+
+  start(printTimeCallback) {
+    this.intervalId = setInterval(() => {
+      this.currentTime++;
+      if (typeof printTimeCallback === "function") {
+        printTimeCallback();
+      };
+    },1000);
+
+    this.millisecondsId = setInterval(() => {
+      if (this.milliseconds>1000) {
+        this.milliseconds = 0;
+      };
+      this.milliseconds++;
+      if (typeof printTimeCallback === "function") {
+        printTimeCallback();
+      }
+    },1);
+  }
+
+  getMinutes() {
+    return Math.floor(this.currentTime/60);
+  }
+
+  getSeconds() {
+    return this.currentTime%60;
+  }
+
+  getMilliseconds() {
+    return this.milliseconds;
+  }
+
+  computeTwoDigitNumber(value) {
+    if (value === 0) {
+      return "00"
+    }
+    else if (value.toString().length === 1) {
+      return `0${value}`
+    }  
+    else {
+      return value.toString()
+    };
+  }
+
+  stop() {
+    clearInterval(this.intervalId);
+    clearInterval(this.millisecondsId);
+  }
+
+  reset() {
+    this.currentTime = 0;
+    this.milliseconds = 0;
+  }
+
+  split() {
+    return `${this.computeTwoDigitNumber(this.getMinutes())}:${this.computeTwoDigitNumber(this.getSeconds())}:${this.computeTwoDigitNumber(this.getMilliseconds())}`
+  }
```

Comment:
This long lines somes can get messy, just keep this in mind (at first i solved exactly like the way you did)

example: 

 ```
 let minutes = this.computeTwoDigitNumber(this.getMinutes())
 let seconds = this.computeTwoDigitNumber(this.getSeconds())

 return `${minutes}:${seconds}`
```